src/main/java/org/jenkinsci/remoting/protocol/IOHub.java

```diff
@@ -292,6 +292,10 @@ public final void removeInterestConnect(SelectionKey key) {
      * @param key the key.
      */
     public final void addInterestRead(SelectionKey key) {
+        if (LOGGER.isLoggable(Level.FINEST)) {
```

Comment:
I wouldn't have added the if here. It's adding verbosity for no reason in that case. I suppose you added that because of the TODO below...